### DISCUSSION TITLE
Forward rules?

### DISCUSSION THREAD

**Discussion by gisborne** [cozodb/cozo#61] [Q&A]
Along with "triggers", I would be looking for forward rules — rules which automatically assert their consequent when a new antecedent becomes true.

Any chance?

**Comment by infogulch** [comment]
Maybe one of these is what you want:

* [Indices](https://docs.cozodb.org/en/latest/stored.html#indices)
* [Triggers](https://docs.cozodb.org/en/latest/stored.html#triggers)

I don't see it mentioned in the docs: Can you add an index to a rule? It'd be like a trigger in some ways, but maybe an index on a rule would be easier to maintain than a trigger. ... On further inspection it seems there are no schema-level 'named rules', just ones defined immediately in the current query. (Correct?) I'm thinking of named rules in the sense of views in sql; I think that analogy holds.
